migrations/src/main/scala/za/co/absa/enceladus/migrations/continuous/migrate01/DatasetMigrator.scala

```diff
@@ -0,0 +1,157 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.migrations.continuous.migrate01
+
+import java.time.ZonedDateTime
+
+import org.mongodb.scala.MongoDatabase
+import org.slf4j.{Logger, LoggerFactory}
+import za.co.absa.enceladus.migrations.continuous.EntityVersionMap
+import za.co.absa.enceladus.migrations.framework.ObjectIdTools
+import za.co.absa.enceladus.migrations.migrations.MigrationToV1.convertConformanceRule
+import za.co.absa.enceladus.migrations.migrations.model0.Serializer0
+import za.co.absa.enceladus.migrations.migrations.model1.Serializer1
+import za.co.absa.enceladus.migrations.migrations.{model0, model1}
+
+import scala.util.{Failure, Success, Try}
+import scala.util.control.NonFatal
+
+/**
+  * The class provides datasets continuous migration from model version 0 to model version 1.
+  *
+  * @param evm         An entity mapper for tracking the mapping between old versions of entities and new ones.
+  * @param databaseOld An instance of a MongoDB database connection containing old model documents.
+  * @param databaseNew An instance of a MongoDB database connection containing new model documents.
+  */
+final class DatasetMigrator(evm: EntityVersionMap,
+                            databaseOld: MongoDatabase,
+                            databaseNew: MongoDatabase) extends EntityMigrator(databaseOld, databaseNew) {
+  private val log: Logger = LoggerFactory.getLogger(this.getClass)
+
+  override protected val collectionBase: String = EntityMigrator.datasetCollection
+
+  /**
+    * Migrates a single instance of dataset.
+    *
+    * @param srcDatasetJson A JSON representation of a dataset in the model 0 format.
+    * @param objectId       An Object Id of the dataset.
+    * @param repo           An entity repository.
+    */
+  def migrateEntity(srcDatasetJson: String, objectId: String, repo: EntityRepository): Unit = {
+    Try {
+      val fixJson = srcDatasetJson.replaceAll("\"jsonClass\" :", "\"_t\" :")
+      val dataset0 = Serializer0.deserializeDataset(fixJson)
+
+      model1.Dataset(
+        dataset0.name,
+        dataset0.version,
+        None,
+        dataset0.hdfsPath,
+        dataset0.hdfsPublishPath,
+        dataset0.schemaName,
+        evm.getSafeVersion(EntityMigrator.schemaCollection, dataset0.schemaName, dataset0.schemaVersion),
+        ZonedDateTime.now(),
+        migrationUserName,
+        ZonedDateTime.now(),
+        migrationUserName,
+        conformance = dataset0.conformance.map(migrateConformanceRule)
+      )
+    } match {
+      case Success(dataset1) =>
+        if (repo.doesDocumentExist(dataset1.name, dataset1.version)) {
+          resolveConflict(dataset1, objectId, repo)
+        } else {
+          normalInsert(dataset1, objectId, repo)
+        }
+      case Failure(e) =>
+        log.warn(s"Encountered a serialization error for '$collectionBase': ${e.getMessage}")
```

Comment:
Warning or error here?

Reply:
Agree, having `error` here is better.

Normal migration encountered several documents that were just bogus, so it was logging these instances as warnings. The continuous migration should be a little stricter maybe.